File: frontend/types/auction.ts
export interface Auction {
  id: number;
  title: string;
  description: string;
  imageUrl: string | null;
  images: AuctionImage[];
  startingPrice: number;
  currentPrice: number;
  reservePrice: number | null;
  startTime: string;
  endTime: string;
  status: "Active" | "Closed" | "Cancelled" | "Draft";
  slug: string;
  sellerId: number;
  sellerUsername: string;
  categoryId: number;
  categoryName: string;
  categorySlug: string;
  bidCount: number;
}

export interface AuctionImage {
  id: number;
  url: string;
  sortOrder: number;
}

export interface AuctionSummary {
  id: number;
  title: string;
  imageUrl: string | null;
  currentPrice: number;
  endTime: string;
  status: string;
  slug: string;
  categoryName: string;
  categorySlug: string;
  bidCount: number;
}

export interface CreateAuctionData {
  title: string;
  description: string;
  imageUrl?: string | null;
  imageUrls?: string[];
  startingPrice: number;
  reservePrice?: number | null;
  endTime: string;
  categoryId: number;
}

export interface UpdateAuctionData {
  title?: string | null;
  description?: string | null;
  imageUrl?: string | null;
  reservePrice?: number | null;
  endTime?: string | null;
  categoryId?: number | null;
}

export interface Category {
  id: number;
  name: string;
  slug: string;
  auctionCount: number;
}

export interface AuctionContact {
  counterpartyUsername: string;
  counterpartyEmail: string;
  viewerRole: "Buyer" | "Seller";
}

export interface WatchlistItem {
  id: number;
  auctionId: number;
  auctionTitle: string;
  auctionSlug: string;
  auctionImageUrl: string | null;
  currentPrice: number;
  endTime: string;
  status: string;
  addedAt: string;
}

export interface DashboardStats {
  totalUsers: number;
  totalAuctions: number;
  activeAuctions: number;
  totalBids: number;
  totalRevenue: number;
  recentBidActivity: BidActivity[];
  recentSales: RecentSale[];
}

export interface RecentSale {
  id: number;
  title: string;
  slug: string;
  currentPrice: number;
  endTime: string;
  sellerUsername: string;
}

export interface BidActivity {
  date: string;
  bidCount: number;
  totalAmount: number;
}

File: backend/BidZone.Domains/Entities/Auction.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BidZone.Domains.Entities;

public class Auction
{
    public int Id { get; set; }

    [Required, MaxLength(200)]
    public string Title { get; set; } = string.Empty;

    [Required]
    public string Description { get; set; } = string.Empty;

    [MaxLength(500)]
    public string? ImageUrl { get; set; }

    [Column(TypeName = "decimal(18,2)")]
    public decimal StartingPrice { get; set; }

    [Column(TypeName = "decimal(18,2)")]
    public decimal CurrentPrice { get; set; }

    [Column(TypeName = "decimal(18,2)")]
    public decimal? ReservePrice { get; set; }

    public DateTime StartTime { get; set; }

    public DateTime EndTime { get; set; }

    [Required, MaxLength(20)]
    public string Status { get; set; } = "Active"; // Active, Closed, Cancelled, Draft

    [MaxLength(255)]
    public string Slug { get; set; } = string.Empty;

    public int SellerId { get; set; }
    public User Seller { get; set; } = null!;

    public int CategoryId { get; set; }
    public Category Category { get; set; } = null!;

    [ConcurrencyCheck]
    public Guid ConcurrencyStamp { get; set; } = Guid.NewGuid();

    public ICollection<Bid> Bids { get; set; } = new List<Bid>();
    public ICollection<AuctionImage> Images { get; set; } = new List<AuctionImage>();
    public ICollection<WatchlistItem> WatchlistItems { get; set; } = new List<WatchlistItem>();
}
